backend/main.go

```diff
@@ -98,8 +98,10 @@ func main() {
 	defer db.Close()
 	h := &Handler{DB: db, Logger: e.Logger}
 	api := e.Group("/api")
-	api.GET("/posts", h.GetPosts)
-	api.POST("/posts", h.CreatePost)
+	apiPosts := api.Group("/posts")
+	apiPosts.GET("", h.GetPosts)
+	apiPosts.GET("/count",h.Count)
+	api.POST("", h.CreatePost)
```

Comment:
ここ`apiPosts.POST`じゃないですかね？
これだと/posts配下のパスじゃなくなる気がします

```suggestion
	apiPosts.POST("", h.CreatePost)
```